elytron/WFLY-16532-additional-scope-for-auth-request.adoc

```diff
@@ -0,0 +1,125 @@
+
+* Adding configurations to the `elytron-oidc-client` subsystem under the `secure-deployment` resource.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-16532[WFLY-16532]
+
+=== Related Issues
+
+* N/A
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [x] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+* WildFly
+
+* WildFly Elytron
+
+=== Other Interested Projects
+
+N/A
+
+=== Relevant Installation Types
+
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* A new attribute named `scope` will be added to the `secure-deployment` resource under the `elytron-oidc-client` subsystem, which will be used 
+to specify additional scope values. These values will be used by the Elytron HTTP OIDC authentication mechanism.  
+
+* It must be possible to configure this attribute using cli commands. For example: 
+
+```
+    /subsystem=elytron-oidc-client/secure-deployment=my-secure-deployment:write-attribute(name=scope, value="openid, offline_access")
+```
+
+* It must also be configured by specifying it in the deployment. This can be done using the `oidc.json` file inside the `WEB-INF` directory. For example: 
+
+```
+    "scope" : "myClient, myclient@redhat.com, offline_access, openid"
+```
+
+* The OpenID Connect Specifications contain more details on https://openid.net/specs/openid-connect-core-1_0.html#ScopeClaims[optional scope values] and https://openid.net/specs/openid-connect-core-1_0.html#OfflineAccess[using scope values to requst Offline Access.]
+
+* Scope values are to be sent as a list of comma separated values inside quotes as seen in the examples above.  
+
+=== Nice-to-Have Requirements
+
+N/A
+
+=== Non-Requirements
+
+N/A
+
+=== Backwards Compatibility
+
+N/A
+
+=== Default Configuration
+
+The `scope` attribute would be undefined by default and in that case, the scope value
+would be hardcoded as `scope=openid` as before. 
+
+== Test Plan
+
+* WildFly Elytron test suite: Test cases implemented for functionality. 
+
+* WildFly test suite: Ensuring the correct scope if chosen and used when the `scope` attribute is 
+changed. 
+
+* Tests will be added for the case where the scope is specified in the subsystem configuration and for the case where it is specified in the deployment configuration. 
+
+== Community Documentation
+
+Documentation for the new scope option will be added to https://github.com/wildfly/wildfly/blob/main/docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Elytron_OIDC_Client.adoc[Elytron OpenID Connect Client Subsystem Configuration].
```

Comment:
Also need to add docs on the WildFly docs: https://docs.wildfly.org/26.1/wildscribe/subsystem/elytron-oidc-client/secure-deployment/index.html

Reply:
The wildfly/docs directory contains the source for the documentation that appears on wildfly.org.

So only the Elytron_OIDC_Client.adoc needs to be updated and then the website will automatically get updated once a new WildFly version is released.

Reply:
Oh, I see. Thank you! 
Please ignore this comment then.